packages/block-library/src/latest-comments/edit.js

```diff
@@ -25,83 +24,59 @@ const MIN_COMMENTS = 1;
  */
 const MAX_COMMENTS = 100;
 
-class LatestComments extends Component {
-	constructor() {
-		super( ...arguments );
+export default function LatestComments( { attributes, setAttributes } ) {
+	const {
+		commentsToShow,
+		displayAvatar,
+		displayDate,
+		displayExcerpt,
+	} = attributes;
 
-		this.setCommentsToShow = this.setCommentsToShow.bind( this );
-
-		// Create toggles for each attribute; we create them here rather than
-		// passing `this.createToggleAttribute( 'displayAvatar' )` directly to
-		// `onChange` to avoid re-renders.
-		this.toggleDisplayAvatar = this.createToggleAttribute(
-			'displayAvatar'
-		);
-		this.toggleDisplayDate = this.createToggleAttribute( 'displayDate' );
-		this.toggleDisplayExcerpt = this.createToggleAttribute(
-			'displayExcerpt'
-		);
-	}
-
-	createToggleAttribute( propName ) {
-		return () => {
-			const value = this.props.attributes[ propName ];
-			const { setAttributes } = this.props;
-
-			setAttributes( { [ propName ]: ! value } );
-		};
-	}
-
-	setCommentsToShow( commentsToShow ) {
-		this.props.setAttributes( { commentsToShow } );
-	}
-
-	render() {
-		const {
-			commentsToShow,
-			displayAvatar,
-			displayDate,
-			displayExcerpt,
-		} = this.props.attributes;
-
-		return (
-			<>
-				<InspectorControls>
-					<PanelBody title={ __( 'Latest comments settings' ) }>
-						<ToggleControl
-							label={ __( 'Display avatar' ) }
-							checked={ displayAvatar }
-							onChange={ this.toggleDisplayAvatar }
-						/>
-						<ToggleControl
-							label={ __( 'Display date' ) }
-							checked={ displayDate }
-							onChange={ this.toggleDisplayDate }
-						/>
-						<ToggleControl
-							label={ __( 'Display excerpt' ) }
-							checked={ displayExcerpt }
-							onChange={ this.toggleDisplayExcerpt }
-						/>
-						<RangeControl
-							label={ __( 'Number of comments' ) }
-							value={ commentsToShow }
-							onChange={ this.setCommentsToShow }
-							min={ MIN_COMMENTS }
-							max={ MAX_COMMENTS }
-							required
-						/>
-					</PanelBody>
-				</InspectorControls>
-				<Disabled>
-					<ServerSideRender
-						block="core/latest-comments"
-						attributes={ this.props.attributes }
+	return (
+		<>
+			<InspectorControls>
+				<PanelBody title={ __( 'Latest comments settings' ) }>
+					<ToggleControl
+						label={ __( 'Display avatar' ) }
+						checked={ displayAvatar }
+						onChange={ () =>
```

Comment:
I believe you could keep the generic toggle function, since all three toggle on change like:

```js
const toggleAttribute = ( propName ) => () =>
	setAttributes( { [ propName ]: ! attributes[ propName ] } );
```

Reply:
I'm not convinced it would make much of a difference, performance-wise.

Reply:
It's more about reusability and not performance.